### DISCUSSION TITLE
DV - Gagal membuat manual legend di ggplot dengan scale_color_manual

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#55] [Q&A] [answered]
Saya ada grafik berikut
![000022](https://user-images.githubusercontent.com/95233522/163975525-13a1a88f-8990-4110-9cea-14387140e9cf.png)
Saya mencoba menambahkan legend dengan code berikut namun gagal: 
```
ggplot(timelines, aes(x = date, y = followers))+
  geom_line(color="red")+
  geom_point(data = timelines, 
             aes(y=streams, 
                 x=date
                 ),
             color="green")+
  geom_point(data = timelines, 
             aes(y=listeners, 
                 x=date
                 ),
             color="blue")+
  theme(axis.title.y = element_blank(),
        legend.position = "top")+
  scale_color_manual(name="Spotify Metrics",
                     breaks=c("Followers","Streams","Listeners"),
                     values=c("Followers"="red","Streams"="green", "Listeners"="blue"))
```

Kenapa legendnya gagal muncul ya? 
Mohon bantuannya, terima kasih.

file csv:
[timelines.csv](https://github.com/teamalgoritma/community/files/8512189/timelines.csv)

**Comment by tomytjandra** [comment, marked as answer]
Halo Pak @Hertzspr, legend tidak muncul karena fungsi `scale_color_manual()` mencari values yang ada di `aes()` setiap geomnya. Dikarenakan parameter `color` tidak diletakkan di dalam `aes()`, maka fungsi tersebut tidak mengenali istilah "color".

Kita dapat menggunakan trick berikut, di mana kita mendefinisikan parameter color di dalam fungsi `aes()`:

```
ggplot(data = timelines, aes(x = date, y = followers)) +
  geom_line(aes(color="Followers")) +
  geom_point(aes(y=streams, color="Streams")) +
  geom_point(aes(y=listeners, color="Listeners")) +
  theme(axis.title.y = element_blank(),
        legend.position = "top") +
  scale_colour_manual(
    name="Spotify Metrics",
    values=c("Followers"="red", "Streams"="green", "Listeners"="blue"))
```

**Notes:** perhatikan juga bahwa saya tidak mendefinisikan secara berulang parameter `data` maupun `x` pada `geom_line()` dan `geom_point()` karena semua parameter yang didefinisikan pada fungsi `ggplot()` berlaku secara **global** untuk keseluruhan geom.

**Output:**
![image](https://user-images.githubusercontent.com/48079239/163991791-809abfaf-783a-41eb-b555-a5f8a3bd4a9e.png)

**Disclaimer:** Cara di atas hanyalah "hack" dan bukan best practice. Practice yang baik adalah kita melakukan reshaping data `timelines` menjadi longer dataframe, sehingga kolom `followers`, `listeners`, dan `streams` terkumpul menjadi satu kolom (misal dinamakan sebagai `metric`. Setelah itu, baru kita lakukan visualisasi dengan `color = metric`. Berikut contoh code reshapingnya, untuk visualisasi silahkan coba lanjutkan secara mandiri ya pak 😄 

```
library(tidyr)
timelines_long <- pivot_longer(timelines, cols = -date, names_to = "metric")
```

<img width="351" alt="image" src="https://user-images.githubusercontent.com/48079239/163992158-975038f4-22e8-4305-991a-2fc3036a3112.png">

**Reply by Hertzspr** [reply]
Terima kasih mas Tomy

**Reply by tomytjandra** [reply]
sama-sama pak @Hertzspr 👍🏻
